docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,380 @@
+
+See the Hatch docs for more information on specifying [dependencies](https://hatch.pypa.io/latest/config/dependency/).
+
+## Environments
+
+A common problem when working with Python code, is managing multiple projects with different dependencies.
+For instance, if we had another app that used version `0.40.0` of Textual, it *may* break if we installed version `0.47.1`.
+
+The standard way of solving this is with *virtual environments* (or *venvs*), which allow each project to have its own set of dependencies.
+Hatch can create virtual environments for us, and makes working with them very easy.
+
+To create a new virtual environment, navigate to the directory with the `pyproject.toml` file and run the following command (you will only need this on first run):
+
+```bash
+hatch env create
+```
+
+Then run the following command to active the virtual environment:
+
+```bash
+hatch shell
+```
+
+If you were to run `python` now, it will have our app and dependencies available for import:
+
+```
+$ python
+Python 3.11.1 (main, Jan  1 2023, 10:28:48) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
+Type "help", "copyright", "credits" or "license" for more information.
+>>> from textual_calculator import calculator
+```
+
+### Running the app
+
+You can launch the calculator from the command line with the following command:
+
+```bash
+python -m textual_calculator.calculator
+```
+
+The `-m` switch tells Python to import the module and run it.
+
+Although you can run your app this way (and it is fine for development), its not ideal for sharing.
+It would be preferable to have a single command to launch the app, so the user can easily run it from the command line.
+To do that, we will need to add an *entry point* to pyproject.toml
+
+## Entry points
+
+An entry point is a function in your project that can be run from the command line.
+For our calculator example, we first need to create a function that will run the app.
+Add the following file to the `src/textual_calculator` folder, and name it `entry_points.py`:
+
+```python
+from textual_calculator.calculator import CalculatorApp
+
+
+def calculator():
+    app = CalculatorApp()
+    app.run()
+```
+
+!!! tip
+
+    If you already have a function that runs your app, you may not need an `entry_points.py` file.
+
+Then edit `pyproject.toml` to add the following section:
+
+```toml
+[project.scripts]
+calculator = "textual_calculator.entry_points:calculator"
+```
+
+Each entry in the `[project.scripts]` section (there can be more than one) maps a command on to an import and function name.
+In the second line above, before the `=` character, `calculator` is the name of the command.
+The string contains the name of the import (`textual_calculator.entry_points`) followed by a colon (`:`) and the name of a function (also called `calculator`).
+
+It is equivalent to doing the following from the Python REPL:
```

Comment:
“It”, what?

“Specifying an entry point like this is equivalent to running the following from the Python REPL:”?